javascript/src/stores/streams/StreamsStore.ts

```diff
@@ -16,16 +14,116 @@ interface Stream {
     createdAt: number;
 }
 
-var StreamsStore = {
-    listStreams(): JQueryPromise<Array<Stream>> {
-        var url = jsRoutes.controllers.api.StreamsApiController.listStreams().url;
-        var promise = $.getJSON(url);
-        promise.fail((jqXHR, textStatus, errorThrown) => {
-            UserNotification.error("Loading streams failed with status: " + errorThrown,
-                "Could not load streams.");
-        });
-        return promise;
-    },
+interface TestMatchResponse {
+    matches: boolean;
+    rules: any;
+}
+
+interface Callback {
+    (): void;
+}
+
+class StreamsStore {
+    private callbacks: Array<Callback> = [];
+
+    listStreams() {
+        return $.getJSON(jsRoutes.controllers.api.StreamsApiController.list().url);
```

Comment:
The only thing that raised a flag in my head is that before we were adding a failure callback in the promise and now is gone. Regarding renaming it, it's okay leaving the refactoring for later as long as we don't change how it behaved before.